pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -175,67 +166,127 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     if (currentAssignment.equals(targetAssignment)) {
       LOGGER.info("Table: {} is already balanced", tableNameWithType);
       if (reassignInstances) {
-        return new RebalanceResult(RebalanceResult.Status.DONE, "Instance reassigned, table is already balanced",
-            instancePartitionsMap, targetAssignment);
+        if (dryRun) {
+          return new RebalanceResult(RebalanceResult.Status.DONE,
+              "Instance reassigned in dry-run mode, table is already balanced", instancePartitionsMap,
+              targetAssignment);
+        } else {
+          return new RebalanceResult(RebalanceResult.Status.DONE, "Instance reassigned, table is already balanced",
+              instancePartitionsMap, targetAssignment);
+        }
       } else {
         return new RebalanceResult(RebalanceResult.Status.NO_OP, "Table is already balanced", instancePartitionsMap,
             targetAssignment);
       }
     }
 
     if (dryRun) {
-      LOGGER.info("Rebalance table: {} in dry-run mode, returning the target assignment", tableNameWithType);
+      LOGGER.info("Rebalancing table: {} in dry-run mode, returning the target assignment", tableNameWithType);
       return new RebalanceResult(RebalanceResult.Status.DONE, "Dry-run mode", instancePartitionsMap, targetAssignment);
     }
 
-    int minAvailableReplicas;
-    if (rebalanceConfig.getBoolean(RebalanceConfigConstants.DOWNTIME, RebalanceConfigConstants.DEFAULT_DOWNTIME)) {
-      minAvailableReplicas = 0;
+    if (!downtime && !currentIdealState.isEnabled()) {
+      LOGGER.warn("Table: {} is disabled, rebalancing it with downtime", tableNameWithType);
```

Comment:
(1) If the table is disabled, it is already down, so maybe just say that we are moving segment assignment?
(2) Why is this a warning? it can be an info.
(3) We should return this information in the status as well.

Reply:
(1) rebalance with downtime is the same as moving segment assignment (added javadoc)
(2) It is a warning because we log it only when downtime is set to false
(3) Added downtime info into the response